packages/prompts/src/index.ts

```diff
@@ -169,24 +169,21 @@ export const confirm = (opts: ConfirmOptions) => {
 };
 
 type Primitive = Readonly<string | boolean | number>;
-interface Option<Value extends Primitive> {
-	value: Value;
-	label?: string;
-	hint?: string;
-}
-export interface SelectOptions<Options extends Option<Value>[], Value extends Primitive> {
+
+type Option<Value> = Value extends Primitive
+	? { value: Value; label?: string; hint?: string }
```

Comment:
Suppose we could extract aliases for these, if it would make easier to decipher violations?

```ts 
type PrimitiveOption<Value extends Primitive> = { value: Value; label?: string; hint?: string>

type ComplexOption<Value> = { value: Value; label: string; hint?: string>
```

Reply:
@natemoo-re thoughts?